presto-parser/src/main/java/com/facebook/presto/sql/tree/QuantifiedComparisonExpression.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.tree;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class QuantifiedComparisonExpression
+        extends Expression
+{
+    public enum Quantifier
+    {
+        ALL,
+        ANY,
+        SOME,
+    }
+
+    private final ComparisonExpression.Type comparisonType;
+    private final Quantifier quantifier;
+    private final Expression left;
+    private final Expression subquery;
+
+    public QuantifiedComparisonExpression(ComparisonExpression.Type comparisonType, Quantifier quantifier, Expression left, Expression subquery)
+    {
+        this(Optional.empty(), comparisonType, quantifier, left, subquery);
+    }
+
+    public QuantifiedComparisonExpression(NodeLocation location, ComparisonExpression.Type comparisonType, Quantifier quantifier, Expression left, Expression subquery)
+    {
+        this(Optional.of(location), comparisonType, quantifier, left, subquery);
+    }
+
+    private QuantifiedComparisonExpression(Optional<NodeLocation> location, ComparisonExpression.Type comparisonType, Quantifier quantifier, Expression left, Expression subquery)
+    {
+        super(location);
+        this.comparisonType = requireNonNull(comparisonType, "comparisonType is null");
+        this.quantifier = requireNonNull(quantifier, "quantifier is null");
+        this.left = requireNonNull(left, "left is null");
+        this.subquery = requireNonNull(subquery, "subquery is null");
+    }
+
+    public ComparisonExpression.Type getComparisonType()
+    {
+        return comparisonType;
+    }
+
+    public Quantifier getQuantifier()
+    {
+        return quantifier;
+    }
+
+    public Expression getLeft()
```

Comment:
Call this getValue() or getExpression(). "left" is weird, since there's no corresponding "right".